### PR TITLE
Copter:Mission Restarts in auto mode

### PR DESCRIPTION
**Pull request by arshPratap**
This PR attempts to solve the issue raised in #16805.The user can arm and restart a mission after successful completion of a mission.Thanks !
As suggested [here](https://github.com/ArduPilot/ardupilot/issues/16805) following the same steps:-
1.)created a mission with wp editor
wp_editor
2.)make sure auto_options=3
auto_options
3.)switched to auto mode
auto
4.)arm throttle(1st time)
arm1
5.)arm throttle (2nd time)
arm2

### CODE REVIEW

**Comment by arshPratap** [comment]
Wanted to know the thoughts of the devs on this issue thats why used a draft for this one. Thanks a lot in advance

File: ArduCopter/mode_auto.cpp [hunk 1]
```diff
@@ -250,11 +253,6 @@ void ModeAuto::land_start(const Vector3f& destination)
 
     // optionally deploy landing gear
     copter.landinggear.deploy_for_landing();
-
-#if AC_FENCE == ENABLED
-    // disable the fence on landing
-    copter.fence.auto_disable_fence_for_landing();
-#endif
```
Comment (peterbarker) [inline ArduCopter/mode_auto.cpp:257]: We've lost this.  Not good.

File: ArduCopter/mode_auto.cpp [hunk 2]
```diff
@@ -1518,9 +1516,8 @@ bool ModeAuto::verify_takeoff()
     // have we reached our target altitude?
     const bool reached_wp_dest = copter.wp_nav->reached_wp_destination();
 
-    // if we have reached our destination
+    // retract the landing gear
     if (reached_wp_dest) {
-        // retract the landing gear
         copter.landinggear.retract_after_takeoff();
```
Comment (peterbarker) [inline ArduCopter/mode_auto.cpp:1521]: Not a good change.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Random whitespace changes aren't fun, sorry :-)

**Comment by rmackay9** [comment]
I think PR https://github.com/ArduPilot/ardupilot/pull/16808 is probably the way to go so we might not want to keep going with this PR.  It's a good attempt though!

**Comment by arshPratap** [comment]
@peterbarker,@rmackay9 thanks for looking into this.I will start working on the suggestions ASAP

**Comment by rmackay9** [comment]
I think we can close this because we've merged https://github.com/ArduPilot/ardupilot/pull/16808

**Pull request closed without merging**